src/utils/config.py

```diff
@@ -142,10 +149,12 @@ def _fetch_global_config(config_url):
         return yaml.safe_load(res.read())
 
 
-def _get_comma_delimited_env(key):
+def _get_comma_delimited_env(key: str) -> Optional[set]:
```

Comment:
whats the benefit of adding this typing

Reply:
* Mainly it's documentation for other coders so they better know what the params and return values are. Of course this could go in a docstring. But the advantage of having real types is that they are syntax validated, and they are checked by mypy
* Type checking by mypy. I find mypy is not super smart and this is not super useful. But to me the documentation advantage is big enough and the extra mypy checking is a free bonus

I like function signature types in python, and I think they're easy to add. I'm trying to move towards using them everywhere